src/main/java/htsjdk/samtools/reference/FastaSequenceFile.java

```diff
@@ -56,9 +60,21 @@ public FastaSequenceFile(final File file, final boolean truncateNamesAtWhitespac
     public FastaSequenceFile(final Path path, final boolean truncateNamesAtWhitespace) {
         super(path);
         this.truncateNamesAtWhitespace = truncateNamesAtWhitespace;
+        this.seekableStream = null;
         this.in = new FastLineReader(IOUtil.openFileForReading(path));
     }
 
+    /**
+     * Constructs a FastaSequenceFile that reads from the specified stream (which must not be compressed, i.e.
+     * the caller is responsible for decompressing the stream).
+     */
+    public FastaSequenceFile(String source, final SeekableStream seekableStream, SAMSequenceDictionary dictionary, final boolean truncateNamesAtWhitespace) {
```

Comment:
@tomwhite This constructor and the following reset method are untested and have non-trivial interactions with the rest of the class.  Could you add tests that exercise them?

Reply:
Done